docassemble/AssemblyLine/al_document.py

```diff
@@ -452,37 +514,68 @@ class ALDocumentBundle(DAList):
     - title
   optional attribute: enabled
   """
+
+  filename:str
+  title: str
+  elements:List[ALDocument] # or ALDocumentBundle
+  cache: DALazyAttribute # stores cached DAFile output with a per-screen load lifetime
+  enabled:bool # optional
+
   def init(self, *pargs, **kwargs):
     super(ALDocumentBundle, self).init(*pargs, **kwargs)
     self.auto_gather=False
     self.gathered=True
-    # self.initializeAttribute('templates', ALBundleList)
+    self.initializeAttribute('cache', DALazyAttribute)
     
-  def as_pdf(self, key='final', refresh=True):
+  def as_pdf(self, key:str='final', refresh:bool=True) -> DAFile:
+    safe_key = space_to_underscore(key)
+
+    log_if_debug('Calling the as_pdf() method for bundle ' + str(self.title))
+
+    if hasattr(self.cache, safe_key):
+      log_if_debug('Returning cached version of bundle ' + self.title)
+      return getattr(self.cache,  safe_key)
+
     if self.filename.endswith('.pdf'):
       ending = ''
     else:
       ending = '.pdf'
-    pdf = pdf_concatenate(self.as_flat_list(key=key, refresh=refresh), filename=self.filename + ending)
+    files = self.enabled_documents()
+    if len(files) == 1:
+      # This case is simplest--no PDF concatenation needed
```

Comment:
Minor nit, but there is PDF concat, just not at this level (the single file could still be a bundle). It's super minor, but it just confused me for a bit.